src/pages/ProductPage.jsx

```diff
@@ -19,9 +21,25 @@ const ProductPage = () => {
 
   const { productId } = useParams(); // Object Destructuring
 
+  function handleShare() {
+    alert("Share this experience");
+  }
+  function handleSave() {
+    alert("Save this experience");
+  }
+
```

Comment:
This logic to (handle shaare & save) could be moved to the Icon Button itself to keep the product page cleaner

Reply:
okay,  I meant what you just described , i suppose its not 'urgent'. Since the Icon works for now I think it can be left like that.